### ISSUE TITLE
pfmatch tail call passes len, not length and reference to undeclared variable.

### ISSUE THREAD

**Issue by petebristow** [Igalia/pflua#252]
Running b48f16f

```
ship:~/pflua/tools# ./pflua-compile --match "match { ip => foo }"
local cast = require("ffi").cast
return function(self,P,length)
   if length < 14 then return false end
   if cast("uint16_t*", P+12)[0] == 8 then return self.foo(P, len) end
   return false
end
```

Which appears broken, similarly there is a reference to a non existent `filter_str` when loading the code. The diff below appears to fix it locally for me. Perhaps though I'm doing something wrong?

```
diff --git a/lib/pflua/src/pf/backend.lua b/lib/pflua/src/pf/backend.lua
index 3dbfed3..9cdcdb2 100644
--- a/lib/pflua/src/pf/backend.lua
+++ b/lib/pflua/src/pf/backend.lua
@@ -288,7 +288,7 @@ local function serialize(builder, stmt)
    end

    local function serialize_call(expr)
-      local args = { 'P', 'len' }
+      local args = { 'P', 'length' }
       for i=3,#expr do table.insert(args, serialize_value(expr[i])) end
       return 'self.'..expr[2]..'('..table.concat(args, ', ')..')'
    end
diff --git a/lib/pflua/src/pf/match.lua b/lib/pflua/src/pf/match.lua
index f18a1b2..c98bb77 100644
--- a/lib/pflua/src/pf/match.lua
+++ b/lib/pflua/src/pf/match.lua
@@ -303,7 +303,7 @@ function compile(str, opts)
    expr = anf.convert_anf(expr)
    expr = ssa.convert_ssa(expr)
    if opts.source then return backend.emit_match_lua(expr) end
-   return backend.emit_and_load_match(expr, filter_str)
+   return backend.emit_and_load_match(expr)
 end

 function selftest()
```

**Comment by kbara** [comment]
@wingo - thoughts?

**Comment by takikawa** [comment]
This should be fixed now by #253.
